pkg/aws/client.go

```diff
@@ -587,28 +581,36 @@ func (c *awsClient) getSubnetRouteTable(subnetID *string, routeTables []*ec2.Rou
 
 // getSubnetIDs will return the list of subnetsIDs supported for the region picked.
 // It is possible to pass non-empty `describeSubnetsInput` to filter results.
-func (c *awsClient) getSubnetIDs(describeSubnetsInput *ec2.DescribeSubnetsInput) ([]*ec2.Subnet, error) {
-	res, err := c.ec2Client.DescribeSubnets(describeSubnetsInput)
+func (c *awsClient) getSubnetIDs(describeSubnetsInput *ec2.DescribeSubnetsInput) ([]*ec2types.Subnet, error) {
+	res, err := c.ec2Client.DescribeSubnets(context.Background(), describeSubnetsInput)
 	if err != nil {
 		return nil, err
 	}
-	return res.Subnets, nil
+
+	var subnetPointers []*ec2types.Subnet
+	for _, subnet := range res.Subnets {
```

Comment:
I wonder if this convertion is needed, from a quick glance, when we use the subnets later we either dereference the pointer or use `awssdk.ToString`.
As AWS simplified their SDK to return a slice of subnets, instead of a slice of pointers to subnets maybe it's worth adjusting our code if we eventually dereference the pointers.

Reply:
Good point I'll refactor the related functions along with that.